data/trie/patriciaMerkleTrie.go

```diff
@@ -484,14 +484,34 @@ func (tr *patriciaMerkleTrie) ExitSnapshotMode() {
 	tr.trieStorage.ExitSnapshotMode()
 }
 
+func getDbThatContainsHash(trieStorage data.StorageManager, rootHash []byte) data.DBWriteCacher {
+	db := trieStorage.GetSnapshotThatContainsHash(rootHash)
+	if db != nil {
+		return db
+	}
+
+	return trieStorage.GetDbThatContainsHash(rootHash)
+}
+
 // GetSerializedNodes returns a batch of serialized nodes from the trie, starting from the given hash
 func (tr *patriciaMerkleTrie) GetSerializedNodes(rootHash []byte, maxBuffToSend uint64) ([][]byte, uint64, error) {
 	tr.mutOperation.Lock()
 	defer tr.mutOperation.Unlock()
 
+	log.Trace("GetSerializedNodes", "rootHash", rootHash)
 	size := uint64(0)
 
-	newTr, err := tr.recreateFromDb(rootHash)
+	db := getDbThatContainsHash(tr.trieStorage, rootHash)
+	if db == nil {
+		return nil, 0, ErrHashNotFound
+	}
+
+	newTsm, err := NewTrieStorageManagerWithoutPruning(db)
```

Comment:
will this not create a lot of data in storage ? Is the data deleted after some time ?